src/mcp/errors.ts

```diff
@@ -1,6 +1,14 @@
-import { mcpError } from "./util";
+import { commandExistsSync, mcpError } from "./util";
 
 export const NO_PROJECT_ERROR = mcpError(
-  "No active project was found. Use the `set_firebase_directory` command to set the project directory (containing firebase.json).",
+  'No active project was found. Use the `set_firebase_directory` tool to set the project directory to an absolute folder location containing a firebase.json config file. Alternatively, change the MCP server config to add [...,"--dir","/absolute/path/to/project/directory"] in its command-line arguments.',
   "PRECONDITION_FAILED",
 );
+
+export function mcpAuthError() {
+  const cmd = commandExistsSync("firebase") ? "firebase" : "npx -y firebase-tools";
+  return mcpError(`The user is not currently logged into the Firebase CLI, which is required to use this tool. Please instruct the user to execute this shell command to sign in or to configure [Application Default Credentials][ADC] on their machine.
```

Comment:
Should we add a docs link explaining how to set up ADC? IDK if that's actually helpful for AIs.

Reply:
Oh yes I literally intended to that's why it looks like a Markdown link.